model/src/application/interaction/mod.rs

```diff
@@ -160,6 +174,7 @@ impl<'de> Visitor<'de> for InteractionVisitor {
 
     #[allow(clippy::too_many_lines)]
     fn visit_map<V: MapAccess<'de>>(self, mut map: V) -> Result<Self::Value, V::Error> {
+        let mut app_permissions: Option<Permissions> = None;
```

Comment:
Since `app_permissions` is an `Option`, it should be wrapped in another `Option` during the deserialization like the other optional fields.

Looks like this is also missing for the `message` field, it should be fixed in another PR.

```suggestion
        let mut app_permissions: Option<Option<Permissions>> = None;
```

Reply:
doesn't have to as L221 is `app_permissions = map.next_value()?;`, if it was `app_permissions = Some(map.next_value()?);` you'd be correct

Reply:
I am not sure it is needed, the double option should only be necessary for nullable fields.